src/JetsProducer.cc

```diff
@@ -1,21 +1,97 @@
 #include <cp3_llbb/Framework/interface/JetsProducer.h>
 #include <cp3_llbb/Framework/interface/Tools.h>
+#include "DataFormats/Math/interface/deltaR.h"
 
 void JetsProducer::produce(edm::Event& event, const edm::EventSetup& eventSetup) {
 
     edm::Handle<std::vector<pat::Jet>> jets;
     event.getByToken(m_jets_token, jets);
 
-    for (const auto& jet: *jets) {
+    edm::Handle<double> rho_handle;
+     event.getByToken(m_rho_token, rho_handle);
+
+    for (auto jet: *jets) {
         if (! pass_cut(jet))
             continue;
 
-        fill_candidate(jet, jet.genJet());
-
         jecFactor.push_back(jet.jecFactor(0));
         area.push_back(jet.jetArea());
         partonFlavor.push_back(jet.partonFlavour());
         hadronFlavor.push_back(jet.hadronFlavour());
+        // Variables needed for 74X b-jet energy regression as of January 26th 2016
```

Comment:
I guess this should be computed only if `applyRegression` is true, or do we want to compute these everytime in case we want to do it offline?